compiler/ilgen/MethodBuilder.cpp

```diff
@@ -80,16 +86,22 @@ namespace OMR
 
 MethodBuilder::MethodBuilder(TR::TypeDictionary *types, OMR::VirtualMachineState *vmState)
    : TR::IlBuilder(asMethodBuilder(), types),
+   // Note: Any allocations in the following memory region that will get deallocated by C++ destruction semantics along with this MethodBuilder class, should instead
+   // be explicitly cleared / deallocated in the MethodBuilder destructor *before* the region itself is destroyed in order to avoid undefined behaviour from invalid
+   // memory references.
+   _segmentProvider(static_cast<TR::SegmentProvider *>(new(TR::Compiler->persistentAllocator()) TR::SystemSegmentProvider(MEM_SEGMENT_SIZE, TR::Compiler->rawAllocator))),
```

Comment:
I don't see it being required, and I think I may have asked @lmaisons about the [existing one](https://github.com/eclipse/omr/blob/master/compiler/ilgen/TypeDictionary.cpp#L469) in TypeDictionary.cpp, not sure what was the answer.

I'm getting rid of both since you don't seem to see it being required either.